src/NHibernate/Dialect/SybaseSQLAnywhere10Dialect.cs

```diff
@@ -343,6 +343,8 @@ protected virtual void RegisterMiscellaneousFunctions()
 			RegisterFunction("transactsql", new StandardSQLFunction("transactsql", NHibernateUtil.String));
 			RegisterFunction("varexists", new StandardSQLFunction("varexists", NHibernateUtil.Int32));
 			RegisterFunction("watcomsql", new StandardSQLFunction("watcomsql", NHibernateUtil.String));
+			RegisterFunction("truncnum", new StandardSafeSQLFunction("truncnum", 2));
+			RegisterFunction("truncate", new StandardSafeSQLFunction("truncnum", 2));
```

Comment:
I do not think that we should register specific function names for being available in HQL when using their specific dialect, since HQL should be database agnostic. `RegisterFunction("truncnum"` should have been added.